src/main/scala/afenton/bazel/bsp/runner/BazelRunner.scala

```diff
@@ -205,12 +207,43 @@ object BazelRunner:
 
   end BazelRunnerImpl
 
-  case class BazelSources(sources: List[String], buildFiles: List[String])
+  case class BspConfig(
+      scalaVersion: String,
+      scalacOptions: List[String],
+      compileJars: List[Path],
+      sources: List[String],
+      packages: List[String],
+      compileLabel: BazelLabel
+  ):
+    private lazy val jarMap: Map[String, Path] =
+      compileJars.map(p => (p.getFileName.toString, p)).toMap
 
-  object BazelSources:
-    given Decoder[BazelSources] = Decoder.instance { c =>
+    def majorScalaVersion: String = scalaVersion match
```

Comment:
I guess this should be `def majorScalaVersion: Option[MajorVersion]` and have `opaque type MajorVersion = String`